pkg/controllers/nodetemplate/controller.go

```diff
@@ -42,20 +43,26 @@ type Controller struct {
 	kubeClient            client.Client
 	subnetProvider        *subnet.Provider
 	securityGroupProvider *securitygroup.Provider
+	amiProvider           *amifamily.Provider
 }
 
-func NewController(kubeClient client.Client, subnetProvider *subnet.Provider, securityGroups *securitygroup.Provider) corecontroller.Controller {
+func NewController(kubeClient client.Client, subnetProvider *subnet.Provider, securityGroups *securitygroup.Provider, amiprovider *amifamily.Provider) corecontroller.Controller {
 	return corecontroller.Typed[*v1alpha1.AWSNodeTemplate](kubeClient, &Controller{
 		kubeClient:            kubeClient,
 		subnetProvider:        subnetProvider,
 		securityGroupProvider: securityGroups,
+		amiProvider:           amiprovider,
 	})
 }
 
 func (c *Controller) Reconcile(ctx context.Context, nodeTemplate *v1alpha1.AWSNodeTemplate) (reconcile.Result, error) {
 	stored := nodeTemplate.DeepCopy()
 
-	err := multierr.Combine(c.resolveSubnets(ctx, nodeTemplate), c.resolveSecurityGroup(ctx, nodeTemplate))
+	err := multierr.Combine(
+		c.resolveSubnets(ctx, nodeTemplate),
+		c.resolveSecurityGroups(ctx, nodeTemplate),
+		c.resolveAMIs(ctx, nodeTemplate),
+	)
```

Comment:
What about calling this function something like `updateSubnets`, `updateSecurityGroups`, etc.? Considering we are trying to always attempt everything regardless if we error on one so that we perform a best-effort update of each of the status items. I'm not sure that updating in the function obfuscates that much considering we have a sub-reconciler pattern in the codebase that updates things inside of their functions